components/centraldashboard/public/components/resource-chart.js

```diff
@@ -0,0 +1,352 @@
+import '@polymer/iron-ajax/iron-ajax.js';
+import '@polymer/iron-icon/iron-icon.js';
+import '@polymer/iron-icons/iron-icons.js';
+import '@polymer/paper-button/paper-button.js';
+import '@polymer/paper-icon-button/paper-icon-button.js';
+import '@polymer/paper-menu-button/paper-menu-button.js';
+import '@polymer/paper-listbox/paper-listbox.js';
+import '@polymer/paper-item/paper-item.js';
+import '@polymer/paper-styles/element-styles/paper-material-styles.js';
+import 'chartjs-plugin-crosshair';
+
+import {html, PolymerElement} from '@polymer/polymer/polymer-element.js';
+import {Chart} from 'chart.js';
+
+Chart.defaults.global.defaultFontFamily = '"Google Sans", sans-serif';
+
+// Preferred colors for Material Charts
+const LINE_COLORS = [
+    '#1e88e5', // google-blue-600,
+    '#00acc1', // paper-cyan-600
+    '#e91e63', // paper-pink-500
+    '#ff9800', // paper-orange-500
+    '#9ccc65', // paper-light-green-400
+];
+const METRICS = new Set(['node', 'podcpu', 'podmem']);
+const INTERVALS = new Set([
+    'Last5m',
+    'Last15m',
+    'Last30m',
+    'Last60m',
+    'Last180m',
+]);
+const MAX_SERIES = 5;
+const MAX_TOOLTIP_LENGTH = 10;
+
+class ResourceChart extends PolymerElement {
+    static get template() {
+        return html`
+        <style include="iron-flex iron-flex-alignment paper-material-styles">
+            :host {
+                @apply --paper-material-elevation-1;
+                background-color: var(--paper-card-background-color,
+                    var(--primary-background-color));
+                border-radius: 5px;
+                display: inline-block;
+                font-size: 14px;
+                max-width: 400px;
```

Comment:
Can we create a mixin called `--dashboard-card` up at dashboard level, which has these shared styles. Then each card implementation can simply do:

```css
:host {
    /* Foo Bar */
    @apply --dashboard-card;
}
```

Reply:
That's a fantastic idea. Done.